File: backend/Spring/src/main/java/Auctionista/Entities/Auction.java
package Auctionista.Entities;

import java.util.Date;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;


@Document("auctions")
public class Auction {


    @Id
    private String _id;

    private String name;
    private String state;
    private List<Bid> bidHistory;
    private String condition;
    private int purchasePrice;
    private int minimumBid;
    private String seller;
    private String winner;
    private String[] images;
    private String description;
    private String auctionType;
    private Tags tags;
    private Date startDate;
    private Date endDate;
    private int highestBid;




    public String get_id() {
        return this._id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getState() {
        return this.state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public List<Bid> getBidHistory() {
        return this.bidHistory;
    }

    public void setBidHistory(List<Bid> bidHistory) {
        this.bidHistory = bidHistory;
    }

    public String getCondition() {
        return this.condition;
    }

    public void setCondition(String condition) {
        this.condition = condition;
    }

    public int getPurchasePrice() {
        return this.purchasePrice;
    }

    public void setPurchasePrice(int purchasePrice) {
        this.purchasePrice = purchasePrice;
    }

    public int getMinimumBid() {
        return this.minimumBid;
    }

    public void setMinimumBid(int minimumBid) {
        this.minimumBid = minimumBid;
    }

    public String getSeller() {
        return this.seller;
    }

    public void setSeller(String seller) {
        this.seller = seller;
    }

    public String getWinner() {
        return this.winner;
    }

    public void setWinner(String winner) {
        this.winner = winner;
    }

    public String[] getImages() {
        return this.images;
    }

    public void setImages(String[] images) {
        this.images = images;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAuctionType() {
        return this.auctionType;
    }

    public void setAuctionType(String auctionType) {
        this.auctionType = auctionType;
    }

    public Tags getTags() {
        return this.tags;
    }

    public void setTags(Tags tags) {
        this.tags = tags;
    }

    public Date getStartDate() {
        return this.startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return this.endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public int getHighestBid() {
        return this.highestBid;
    }

    public void setHighestBid(int highestBid) {
        this.highestBid = highestBid;
    }

    
    
    



}

File: backend/dotNet/MicroServices/AuctionMicroService/Models/Auction.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson.Serialization.IdGenerators;

namespace AuctionMicroService.Models
{
    public class Auction
    {
        
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }
        public string? Name { get; set; }
        public string? Condition { get; set; }
        public string? AuctionType { get; set; }
        public string? Description { get; set; }
        public string? State { get; set; }
        public int MinimumBid { get; set; }
        public int PurchasePrice { get; set; }
        public int HighestBid { get; set; }
        
        [BsonRepresentation(BsonType.ObjectId)]
        public List<string> Images { get; set; }
        
        [BsonRepresentation(BsonType.ObjectId)]
        public string Seller { get; set; }

        
        [BsonRepresentation(BsonType.ObjectId)]
        public string Winner { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public List<Bid> Bids{ get; set; }
        public int Postage { get; set; }
        
        public Tags? Tags { get; set; }

    }
}
